index.js

```diff
@@ -86,26 +86,43 @@ class MultiCoreIndexer extends TypedEmitter {
   }
 
   /**
-   * Add a core to be indexed
+   * Add a hypercore to the indexer. Must have the same value encoding as other
+   * hypercores already in the indexer.
+   *
+   * Rejects if called after the indexer is closed.
+   *
    * @param {import('hypercore')<T, any>} core
    */
   addCore(core) {
+    this.#assertOpen('Cannot add core after closing')
     const coreIndexStream = new CoreIndexStream(core, this.#createStorage)
     this.#indexStream.addStream(coreIndexStream)
   }
 
   /**
-   * Resolves when indexing state is 'idle'
+   * Resolves when indexing state is 'idle'.
+   *
+   * Resolves if the indexer is closed before this resolves. Rejects if called
+   * after the indexer is closed.
    */
   async idle() {
+    this.#assertOpen('Cannot await idle after closing')
     if (this.#getState().current === 'idle') return
     if (!this.#pendingIdle) {
       this.#pendingIdle = pDefer()
     }
     return this.#pendingIdle.promise
   }
 
+  /**
+   * Stop the indexer and flush index state to storage. This will not close the
+   * underlying storage - it is up to the consumer to do that.
+   *
+   * Rejects if called more than once.
```

Comment:
is this preferred over no-oping when closed already?

Reply:
should update the docs with this note to describe the behavior anyways

Reply:
Double-closing is now a no-op in 56564f62b92980cc07142da3f4f40dcaebf16fb7. Updated the docs in d32e778d04e524ef18805cdede7dbfb78d4ac966.